### ISSUE TITLE
e.SetRequestBodyString() Truncates Anything Longer Than Original Request Body

### ISSUE THREAD

**Issue by clindsb1** [justcoding121/Titanium-Web-Proxy#70]
I'm trying at append some information into a soap header using titanium web proxy. I can get the request body using e.GetRequestBodyAsString() but when I modify the string and send it back using e.SetRequestBodyString() the string is truncated to the length of the original request body. 

For example, if the original request body was:

    <soap:Envelope>
        <soap:Header />
        <soap:Body>
            <tag>
                info
            </tag>
        </soap:Body>
    </soap:Envelope>

And I changed it to be:

    <soap:Envelope>
        <soap:Header>
        </soap:Header>
        <soap:Body>
            <tag>
                info
            </tag>
        </soap:Body>
    </soap:Envelope>

The e.SetRequestBodyString would actually send:

    <soap:Envelope>
        <soap:Header>
        </soap:Header>
        <soap:Body>
            <tag>
                info
            </tag>
        </soap:Body>
    </so

In other words, if the original request was 500 bytes, and the modified message I was trying to send was 512 bytes, the actual message sent with the e.SetRequestBodyString would just be the first 500 bytes of the 512 byte message that I was trying to send. 

Using Titanium Web Proxy 2.1.110 with .Net Framework 4.5

**Comment by clindsb1** [comment]
I figured out that I needed to change the length of the Request by using e.ProxySession.Request.ContentLength = newLength.

**Comment by justcoding121** [comment]
When setting request body the new content length is calculated and should be updated. This is a potential bug.

**Comment by justcoding121** [comment]
Fixed this feature in latest beta release

[Nuget link for beta](https://www.nuget.org/packages/Titanium.Web.Proxy/2.1.111-beta)

**Comment by clindsb1** [comment]
Confirmed fixed.
